src/main/java/org/cactoos/http/HtHeaders.java

```diff
@@ -38,25 +41,27 @@
  * @version $Id$
  * @since 0.1
  */
-public final class HtHeaders extends MapEnvelope<String, String> {
+public final class HtHeaders extends MapEnvelope<String, List<String>> {
 
     /**
      * Ctor.
      * @param head Response head part
      */
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public HtHeaders(final Input head) {
         super(() -> {
             final String[] lines = new TextOf(head).asString().split("\r\n");
-            final Map<String, String> map = new HashMap<>(lines.length - 1);
+            final Map<String, List<String>> map =
+                new HashMap<>(lines.length - 1);
             for (int idx = 1; idx < lines.length; ++idx) {
```

Comment:
@krzyk I think that here you could use cactoos `SplitText` and its `iterator` method.